virt_up/instance.py

```diff
@@ -590,16 +590,27 @@ def address(self):
             self.start()
 
         address_source = self.meta.get('address-source', 'agent')
-        if address_source == 'agent':
-            address = self._address_from_ia(source='agent')
-        elif address_source == 'lease':
-            address = self._address_from_ia(source='lease')
-        elif address_source == 'arp':
-            address = self._address_from_arp()
-        elif address_source == 'dns':
-            address = self._address_from_dns()
-        else:
-            raise ValueError(f"Invalid address_source '{address_source}' in instance '{self.name}'.")
+        for retries in range(120, -1, -1):
+            if address_source == 'agent':
+                address = self._address_from_ia(source='agent')
+            elif address_source == 'lease':
+                address = self._address_from_ia(source='lease')
+            elif address_source == 'arp':
+                address = self._address_from_arp()
+            elif address_source == 'dns':
+                address = self._address_from_dns()
+            else:
+                raise ValueError(f"Invalid address_source '{address_source}' in instance '{self.name}'.")
+
+            if address is not None:
+                host_up = True if os.system(f"ping -c 2 {address}") is 0 else False
```

Comment:
This should be '== 0' not 'is 0'. also we dont need a if:else to set True/False.  e.g.
The rest of this module uses 'sh' to run external commands.

Reply:
ack